spring-boot-project/spring-boot/src/main/java/org/springframework/boot/logging/LoggerGroups.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2012-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.boot.logging;
+
+import java.util.Map;
+import java.util.Set;
+
+import org.springframework.util.Assert;
+
+/**
+ * Common abstraction for logger groups.
+ *
+ * @author HaiTao Zhang
+ * @since 2.2.0
+ */
+public class LoggerGroups {
```

Comment:
Have you considered updating `LoggingSystem` rather than registering another singleton in the context? Given that became a feature handled there, perhaps we should expose it rather than having to lookup another type.